app/models/adviser.rb

```diff
@@ -46,6 +48,10 @@ def field_order
 
   private
 
+  def upcase_postcode
```

Comment:
The transformation occurring before validation is exactly the right time in this instance since the upcasing occurs solely for validation, and we know the object can't be persisted (well, at least not in the usual code path) without first being validated.